src/cpu.c

```diff
@@ -19,4 +19,57 @@ cpu* mgec_new_cpu() {
   c->e = 1;
 
   return c;
+}
+
+#define load(C, R, V) _Generic((V), u8: load8, u16: load16)(C, R, V)
+
+static inline cycles load8(cpu* c, cpu_register_t reg, u8 value) {
+  switch (reg) {
+  case XL:
+    c->xl = value;
+    break;
+  case XH:
+    c->xh = value;
+    break;
+  case YL:
+    c->yl = value;
+    break;
+  case YH:
+    c->yh = value;
+    break;
+  // Making sure we're handling all cases
+  // TODO: some error condition
+  case X:
+  case Y:
+  case A:
+  case B:
```

Comment:
Similarly, due to the `XBA` instruction existing (which eXchanges the `B` and `A` registers in 8-bit accumulator mode), I do not think we should be able to set the `B` register via `load8`. Additionally, the `C` register is only available in 16-bit accumulator and native mode, so it should continue to be unavailable in `load8`.

Reply:
I'm leaving the fallthrough cases here along with the default to A) catch all cases and B) do an error at some point in the future.

Reply:
Don't mind the fallthrough cases, but I think the A register should still be loadable in this function.